libs/ipc/src/item.cpp

```diff
@@ -0,0 +1,38 @@
+#include <random>
+
+#include <fmt/format.h>
+#include <glaze/json.hpp>
+
+#include <tfc/ipc/details/item_glaze_meta.hpp>
+#include <tfc/ipc/item.hpp>
+#include <tfc/progbase.hpp>
+
+namespace tfc::ipc::item {
+
+item make() {
+  // todo how about storing the random engine in shared memory?
+  static std::mt19937_64 random_engine{};  // NOSONAR
+  static bool first_call{ true };
+  if (first_call) {
+    first_call = false;
+    auto proc_id{ std::hash<std::string>{}(fmt::format("{}{}", tfc::base::get_exe_name(), tfc::base::get_proc_name())) };
+    random_engine.seed(proc_id);
+  }
+  uuids::basic_uuid_random_generator<std::mt19937_64> random_generator{ random_engine };
+
+  return { .item_id = random_generator(), .entry_timestamp = std::chrono::system_clock::now() };
```

Comment:
Having the state here global makes testing hard and the function unnecessarily complex.
I would like to see the global state removed from the basic make function
and rather a seed accepted. The global state can then be stored somewhere
else if required but the basic function pure.

Another thing is that I would like us to consider PCG rng instead of the mt
just because of size and initialization time.

Reply:
now using pcg as proposed